api/src/opentrons/protocol_api/execute_v1.py

```
@@ -0,0 +1,234 @@
+# execute v1 and v2 protocols
+import itertools
+import logging
+from typing import Any, Dict, Optional
+
+from .contexts import ProtocolContext, InstrumentContext
+from .back_compat import BCLabware
+from . import labware
+from opentrons.types import Point, Location
+
+MODULE_LOG = logging.getLogger(__name__)
+
+
+def load_pipettes_from_json(
+        ctx: ProtocolContext,
+        protocol: Dict[Any, Any]) -> Dict[str, InstrumentContext]:
+    pipette_data = protocol.get('pipettes', {})
+    pipettes_by_id = {}
+    for pipette_id, props in pipette_data.items():
+        model = props.get('model')
```

If these keys are mandatory we should access them via dict indexing: `model = props['model']` which will fail fast with a `KeyError` if the protocol is malformed and the keys aren't present, and fixes issues with types since `get()` returns `Optional[T]`